packages/core/src/character/controller-local.ts

```diff
@@ -80,159 +83,176 @@ export class LocalController {
-    const characterQuaternion = this.characterModel?.getWorldQuaternion(new Quaternion());
+  updateNetworkState(): void {
+    if (!this.model?.mesh) return;
+    const characterQuaternion = this.model.mesh.getWorldQuaternion(new Quaternion());
     const positionUpdate = new Vector3(
-      this.characterModel?.position.x,
-      this.characterModel?.position.y,
-      this.characterModel?.position.z,
+      this.model.mesh.position.x,
+      this.model.mesh.position.y,
+      this.model.mesh.position.z,
     );
     const rotationUpdate = new Vector2(characterQuaternion?.y, characterQuaternion?.w);
     this.networkState = {
       id: this.id,
       location: positionUpdate,
       rotation: rotationUpdate,
-      state: this.currentAnimation as AnimationState,
+      state: this.model.currentAnimation as AnimationState,
     };
   }
 
-  updateFromNetwork(clientUpdate: ClientUpdate): void {
-    if (!this.characterModel || clientUpdate.id !== this.id) return;
-    const { location, rotation, state } = clientUpdate;
-    this.characterModel.position.x = location.x;
-    this.characterModel.position.y = location.y;
-    this.characterModel.position.z = location.z;
-    const rotationQuaternion = new Quaternion(0, rotation.x, 0, rotation.y);
-    this.characterModel.setRotationFromQuaternion(rotationQuaternion);
-    if (state !== this.currentAnimation) {
-      this.transitionToAnimation(state);
-    }
+  resetPosition(): void {
+    if (!this.model?.mesh) return;
+    this.model.mesh.position.y = 5;
   }
 
   update(inputManager: InputManager, cameraManager: CameraManager, runTime: RunTimeManager): void {
-    if (!this.characterModel || !this.animationMixer) return;
+    if (!this.model?.mesh || !this.model?.animationMixer) return;
     if (!this.thirdPersonCamera) this.thirdPersonCamera = cameraManager.camera;
 
     const movementKeysPressed = inputManager.isMovementKeyPressed();
     const forward = inputManager.isKeyPressed("w");
     const backward = inputManager.isKeyPressed("s");
     const left = inputManager.isKeyPressed("a");
     const right = inputManager.isKeyPressed("d");
+
+    this.jumpInput = inputManager.isJumping();
+
     this.inputDirections = { forward, backward, left, right };
     this.runInput = inputManager.isShiftPressed();
 
     if (movementKeysPressed) {
       const targetAnimation = this.getTargetAnimation();
-      if (targetAnimation !== this.currentAnimation) {
-        this.transitionToAnimation(targetAnimation);
-      }
+      this.model.updateAnimation(targetAnimation, runTime.smoothDeltaTime);
     } else {
-      if (this.animations.idle) this.transitionToAnimation("idle");
+      this.model.updateAnimation("idle", runTime.smoothDeltaTime);
     }
-    this.updatePosition(runTime.smoothDeltaTime);
+
     if (anyTruthness(this.inputDirections)) this.updateRotation();
-    this.animationMixer.update(runTime.smoothDeltaTime);
-    this.getNetworkState();
+
+    for (let i = 0; i < 15; i++) {
```

Comment:
This `15` should probably be constant to avoid changing just one of these two values

Reply:
Sure. Fixed in this line of [commit 5](https://github.com/mml-io/mml-playground/pull/21/commits/01fc1ea646d4c1e9d0eb55bcff3e2de0202b47ff#diff-43be350ff578d058692c98abe33e16d4a551db5800e5ed5c8e12a35718a91fcfR253)